photon-lib/src/main/scala/com/linkedin/photon/ml/Types.scala

```diff
@@ -24,6 +24,16 @@ object Types {
   type SSV = org.apache.spark.mllib.linalg.SparseVector
   type SparkVector = org.apache.spark.mllib.linalg.Vector
 
+  // A "sample" is a training, validation or scoring vector of feature values
+  type UniqueSampleId = Long
+
+  // A "coordinate" is made up of several random effects
```

Comment:
I think that's fine, though the I would cut the `"per-user", "per-song", ...` part since we would need to explain that refers to the random effect coodinates, as well as the factored random effect coordinates, which use but aren't quite a regular matrix factorization coordinate, and so on